contracts/modules/v2/Claims.sol

```diff
@@ -0,0 +1,381 @@
+            block.timestamp >= poll.end +
+            payoutCooldownDays * 1 days +
+            config.payoutRedemptionPeriodDays * 1 days
+          ) {
+            payoutStatus = PayoutStatus.UNCLAIMED;
+          } else {
+            payoutStatus = PayoutStatus.PENDING;
+          }
+        }
+      } else if (claimStatus == ClaimStatus.DENIED) {
+        payoutStatus = PayoutStatus.DENIED;
+      } else {
+        payoutStatus = PayoutStatus.PENDING;
+      }
+    }
+
+    (
+      uint24 productId,
+      uint8 payoutAsset,
+      /*uint96 amount*/,
+      uint32 coverStart,
+      uint32 coverPeriod,
+      /*uint80 nxmPrice*/
+    ) = cover().covers(claim.coverId);
+
+    uint coverEnd = coverStart + coverPeriod;
+
+    string memory assetSymbol;
+    if (claim.payoutAsset == 0) {
+      assetSymbol = "ETH";
+    } else {
+      try IERC20Detailed(pool().assets(claim.payoutAsset)).symbol() returns (string memory v) {
+        assetSymbol = v;
+      } catch {
+        // return assetSymbol as an empty string and use claim.payoutAsset instead in the UI
+      }
+    }
+
+    return ClaimDisplay(
+      id,
+      productId,
+      claim.coverId,
+      claim.amount,
+      assetSymbol,
+      claim.payoutAsset,
+      coverStart,
+      coverEnd,
+      poll.start,
+      poll.end,
+      uint(claimStatus),
+      uint(payoutStatus)
+    );
+  }
+
+  /**
+   *  Returns an array of claims aggregated in a human-friendly format.
+   *
+   *  @dev This view is meant to be used in user interfaces to get claims in a format suitable for
+   *  displaying all relevant information in as few calls as possible. It can be used to paginate
+   *  claims by providing the following paramterers:
+   *
+   *  @param ids   Array of Claim ids which are returned as ClaimDisplay
+   */
+  function getClaimsToDisplay (uint104[] calldata ids)
+  external view returns (ClaimDisplay[] memory) {
+    ClaimDisplay[] memory claimDisplays = new ClaimDisplay[](ids.length);
+    for (uint i = 0; i < ids.length; i++) {
+      uint104 id = ids[i];
+      claimDisplays[i] = getClaimDisplay(id);
+    }
+    return claimDisplays;
+  }
+
+  /* === MUTATIVE FUNCTIONS ==== */
+
+  /**
+   *  Submits a claim for assessment
+   *
+   *  @dev This function requires an ETH submission fee. See: _getSubmissionFee()
+   *
+   *  @param coverId          Cover identifier
+   *  @param requestedAmount  The amount expected to be received at payout
+   *  @param hasProof         When true, a ProofSubmitted event is emitted with ipfsProofHash.
+   *                          When false, no ProofSubmitted event is emitted to save gas if the
+   *                          cover wording doesn't enforce a proof of loss.
+   *  @param ipfsProofHash    The IPFS hash required for proof of loss. It is ignored if hasProof
+   *                          is false
+   */
+  function submitClaim(
+    uint24 coverId,
+    uint96 requestedAmount,
+    bool hasProof,
+    string calldata ipfsProofHash
```

Comment:
can you check if this is empty to avoid passing the extra hasProof flag?

Reply:
seems like the check would be `bytes(string).length > 0`

Reply:
I'm thinking someone could call this function with a valid proof, but the hadProof flag set to false or smth